frontend/src/utils/use_blur.ts

```diff
@@ -0,0 +1,22 @@
+import { useRef } from 'react';
+import { useEvent } from './use_event';
+
+export function useOnBlur(ref: HTMLElement | null, callback: () => void) {
+  // we need this terrible timeout hack because sometimes click events are fired, that are not on
+  // the target. these are immidiately followed by the actual event
```

Comment:
How/when/why does this happen?

Reply:
I am not really sure, why this happens, but it happens whenever I focus the slate editor and then try to operate a dropdown menu for speaker options

Reply:
Also this should probably be called `useDebouncedOnClick` or something, right?

Reply:
Hm... not really, because it detects Clicks outside. maybe `useOnDebouncedOutsideClick`?

Reply:
My dear fried, I hope your foot is well

https://github.com/transcribee/transcribee/blob/9704d73cea481f43ecca9cb65cb23ed3ae725deb/frontend/src/editor/transcription_editor.tsx#L122

Reply:
> Also this should probably be called `useDebouncedOnClick` or something, right?

I vote for `useOnClickOutside`